File: c++/leetcode/461. Hamming Distance.cpp
class Solution
{
public:
  int hammingDistance(int x, int y)
  {
    int ans = 0;
    int _xor = x xor y;

    while (_xor)
    {
      ans += _xor & 1 == 1 ? 1 : 0;
      _xor >>= 1;
    }

    return ans;
  }
};

File: javascript/leetcode/461. Hamming Distance.js
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function (x, y) {
  let ans = 0;
  let _xor = x ^ y;

  while (_xor) {
    ans += _xor & (1 == 1) ? 1 : 0;
    _xor >>>= 1;
  }

  return ans;
};
